### PR TITLE
Create ONVDB client certificate for Octavia

### PR DESCRIPTION
**Pull request by gthiemonge**
The certificate is required by the Octavia OVN provider when TLS is enabled at the Pod level.

Jira: [OSPRH-6065](https://issues.redhat.com//browse/OSPRH-6065)

### CODE REVIEW

File: pkg/openstack/octavia.go [hunk 1]
```diff
@@ -67,6 +69,36 @@ func ReconcileOctavia(ctx context.Context, instance *corev1beta1.OpenStackContro
...
 		instance.Spec.Octavia.Template.OctaviaAPI.TLS = octavia.Spec.OctaviaAPI.TLS
+
+		serviceName := "octavia"
+		// create ovndb client certificate for octavia
+		certRequest := certmanager.CertificateRequest{
+			IssuerName: instance.GetOvnIssuer(),
+			CertName:   fmt.Sprintf("%s-ovndbs", serviceName),
+			Duration:   nil,
+			Hostnames: []string{
+				fmt.Sprintf("%s.%s.svc", serviceName, instance.Namespace),
```
Comment (olliewalsh) [inline pkg/openstack/octavia.go:80]: Should this be a wildcard cert since the pod dns names are not predictable.

File: pkg/openstack/octavia.go [hunk 2]
```diff
@@ -67,6 +69,36 @@ func ReconcileOctavia(ctx context.Context, instance *corev1beta1.OpenStackContro
...
+				fmt.Sprintf("%s.%s.svc.%s", serviceName, instance.Namespace, "cluster.local"),
```
Comment (olliewalsh) [inline pkg/openstack/octavia.go:81]: May be best to move `cluster.local` to a const

Comment (stuggi) [reply pkg/openstack/octavia.go:81]: I have added a const in https://github.com/openstack-k8s-operators/openstack-operator/pull/740

File: pkg/openstack/octavia.go (hunk 1 above)
Comment (gthiemonge) [reply pkg/openstack/octavia.go:80]: I see some services like neutron not using the wildcard: https://github.com/openstack-k8s-operators/openstack-operator/blob/main/pkg/openstack/neutron.go#L73
is it really useful here?

File: pkg/openstack/octavia.go (hunk 2 above)
Comment (gthiemonge) [reply pkg/openstack/octavia.go:81]: ack

**Comment by weinimo** [comment]
recheck
RETRY_LIMIT

**Comment by stuggi** [comment]
needs a rebase

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**